src/rust/engine/rule_graph/src/lib.rs

```diff
@@ -1008,13 +1070,28 @@ impl<R: Rule> RuleGraph<R> {
       .filter_map(|(k, deps)| match k {
         EntryWithDeps::Root(_) => {
           let root_str = entry_with_deps_str(k);
+          let mut dep_entries = deps
+            .all_dependencies()
+            .map(|d| entry_node_str_with_attrs(d))
+            .collect::<Vec<_>>();
+          dep_entries.sort();
```

Comment:
Is it possible to chain `.sort()` onto the end of the initial declaration so that you can avoid marking this as `mut`? (Not a rhetorical question - I'm still fairly green with Rust)

Reply:
In general, you can usually just type "vec rust" into duckduckgo and find the documentation you're looking for (see https://doc.rust-lang.org/std/vec/struct.Vec.html#method.sort)! In this case, I believe out of a predisposition for not unnecessarily allocating, the rust `Vec` methods for sorting all return `()`. Hence the mutability.

This could be rewritten behind a utility method, but this is idiomatic enough rust code (I believe) even if the `mut` could *technically* be unnecessary. @pierrechevalier83 @illicitonion is that mostly right?

Reply:
Yes, `sort` is defined on a `Vec`, and acts inplace, so this is the idiomatic way to do it.

`sort` doesn't exist on an `Iterator` because an `Iterator` could iterate forever and there wouldn't be a good way to sort that:
For example, what would you expect this code to do?
```
    rand::thread_rng().sample_iter(rand_distr::Uniform::new(0, 42)).sort()
```